### ISSUE TITLE
`L029` doesn't honor `ignore_words` in CTEs during pre-commit

### ISSUE THREAD

**Issue by ljhopkins2** [sqlfluff/sqlfluff#2522]
## Expected Behaviour
`ignore_words` for L029 should be respected during pre-commit.

## Observed Behaviour
Rule L029 fails on ignored words during pre-commit.

## Steps to Reproduce
- Note the config for L029 in the Configuration section below.

- Create a SQL file containing this statement:
```sql
with
    test_data as (
        select '2022-01-01' as date, 'abc' as foo
        union all
        select '2022-01-01' as date, 'def' as foo
        union all
        select '2022-01-02' as date, 'ghi' as foo
    ),

    test_data_2 as (
        select '2022-01-01' as date, 'rst' as bar  -- noqa: L029
        union all
        select '2022-01-01' as date, 'uvw' as bar  -- noqa: L029
        union all
        select '2022-01-02' as date, 'xyz' as bar  -- noqa: L029
    )

select test_data.date, test_data.foo, test_data_2.bar
from test_data
inner join test_data_2 on test_data.date = test_data_2.date
```

- Lint the file from the command line:
```
All Finished 📜 🎉!
```

- With pre-commit set up, commit the file, and note the output. `date` keywords inside the CTEs that aren't explicitly flagged as `noqa` fail.
```
== [path/to/file/test.sql] FAIL                               
L:   3 | P:  32 | L029 | Keywords should not be used as identifiers.
L:   5 | P:  32 | L029 | Keywords should not be used as identifiers.
L:   7 | P:  32 | L029 | Keywords should not be used as identifiers.
```

## Dialect
Redshift

## Version
```
sqlfluff, version 0.9.3
sqlfluff-templater-dbt==0.9.3
Python 3.9.9

dbt installed version: 1.0.1
Plugins:
  - redshift: 1.0.0
  - postgres: 1.0.1
```

## Configuration
```
[sqlfluff]
templater = dbt
dialect = redshift
exclude_rules = L003, L018, L036

[sqlfluff:rules]
tab_space_size = 4
indent_unit = space
max_line_length = 90
capitalisation_policy = lower
extended_capitalisation_policy = lower
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
template_blocks_indent = False

[sqlfluff:rules:L029]
ignore_words = date, domain, hash, hostname, language

[sqlfluff:rules:L052]
multiline_newline = True
```

**Comment by tunetheweb** [comment]
Can you give your pre-commit config? This suggests it's using a different sqlfluff config file for that, as ti should just be running `sqlfluff lint` in the same way as the command line.
